_pytest/cacheprovider.py

```diff
@@ -168,6 +170,41 @@ def pytest_sessionfinish(self, session):
             config.cache.set("cache/lastfailed", self.lastfailed)
 
 
+class NFPlugin(object):
+    """ Plugin which implements the --nf (run new-first) option """
+
+    def __init__(self, config):
+        self.config = config
+        self.active = config.option.newfirst
+        self.all_items = config.cache.get("cache/allitems", {})
```

Comment:
I believe the default value here should be `[]`, for consistency more than anything else.